apps/fz_http/test/fz_http_web/controllers/json/rule_controller_test.exs

```diff
@@ -1,64 +1,177 @@
 defmodule FzHttpWeb.JSON.RuleControllerTest do
-  use FzHttpWeb.ConnCase, async: true
-  @moduletag api: true
+  use FzHttpWeb.ApiCase, async: true
 
-  @rule_params %{
-    "destination" => "5.5.5.5/24",
+  @accept_rule_params %{
+    "destination" => "1.1.1.1/24",
     "action" => "accept",
+    "port_type" => "udp",
+    "port_range" => "1 - 2"
+  }
+
+  @drop_rule_params %{
+    "destination" => "5.5.5.5/24",
+    "action" => "drop",
     "port_type" => "tcp",
     "port_range" => "1 - 65000"
   }
 
-  describe "show rule" do
+  describe "[authed] GET /v0/rules/:id" do
+    setup _tags, do: {:ok, conn: authed_conn()}
     setup :create_rule
 
-    test "shows rule", %{api_conn: conn, rule: %{id: id}} do
+    test "shows rule", %{conn: conn, rule: %{id: id}} do
       conn = get(conn, ~p"/v0/rules/#{id}")
       assert %{"id" => ^id} = json_response(conn, 200)["data"]
     end
+
+    test "renders 404 for rule not found", %{conn: conn} do
+      assert_error_sent 404, fn ->
+        get(conn, ~p"/v0/rules/003da73d-2dd9-4492-8136-3282843545e8")
+      end
+    end
   end
 
-  describe "create rule" do
-    @tag params: @rule_params
-    test "creates rule", %{api_conn: conn, unprivileged_user: user, params: params} do
+  describe "[authed] POST /v0/rules" do
+    setup _tags, do: {:ok, conn: authed_conn()}
+
+    @tag params: @accept_rule_params
+    test "creates accept rule when valid", %{conn: conn, params: params} do
+      user = conn.private.guardian_default_resource
+      conn = post(conn, ~p"/v0/rules", rule: Map.merge(params, %{"user_id" => user.id}))
```

Comment:
I feel that it will look much cleaner and more straightforward to read if we just use the module attribute right away, without adding a tag just to match on it one line below to create a variable.

```suggestion
    test "creates accept rule when valid", %{conn: conn} do
      user = conn.private.guardian_default_resource
      conn = post(conn, ~p"/v0/rules", rule: Map.merge(@accept_rule_params, %{"user_id" => user.id}))
```

Reply:
👍

Reply:
Fixed